app/server/publish.js

```diff
@@ -31,6 +31,22 @@ Meteor.publish('getViewStates', function(section, flags) {
 
         return View.find(flags);
     */
+    var userId = this.userId;
+
+    //checks if user is loggedd in
+    if(!userId){
+        throw new Meteor.Error(401, 'you must be logged in!');
```

Comment:
Pwedeng 403 nalang pag di nakalogin.

Check mo din ang this.profile.type kung admin